SYCL/ESIMD/api/functional/operators/operator_decrement_and_increment.hpp

```diff
@@ -0,0 +1,299 @@
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed &= run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed &= run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> source_simd_out(NumElems, allocator);
+    shared_vector<DataT> result_simd_out(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         allocator);
+
+    queue.submit([&](sycl::handler &cgh) {
+      const DataT *const ref = shared_ref_data.data();
+      DataT *source_simd_data_ptr = source_simd_out.data();
+      DataT *result_simd_data_ptr = result_simd_out.data();
+
+      cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(
+                ref, source_simd_data_ptr, result_simd_data_ptr);
+          });
+    });
+    queue.wait_and_throw();
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      DataT expected_source_value = shared_ref_data[i];
+
+      const DataT expected_return_value =
+          TestCaseT::apply_operator(expected_source_value);
+
+      passed &= verify_result(i, expected_source_value, source_simd_out[i],
+                              "unexpected argument modification", data_type);
+      passed &= verify_result(i, expected_return_value, result_simd_out[i],
+                              "unexpected return value", data_type);
+    }
+
+    return passed;
+  }
+
+  bool verify_result(size_t i, DataT expected, DataT retrieved,
+                     const std::string &simd_type,
+                     const std::string &data_type) {
+    bool passed = true;
+    if constexpr (type_traits::is_sycl_floating_point_v<DataT>) {
+      if (std::isnan(expected) && !std::isnan(retrieved)) {
+        passed = false;
+
+        // TODO: Make ITestDescription architecture more flexible.
+        // We are assuming that the NaN opcode may differ
+        std::string log_msg("Failed for simd<");
+        log_msg += data_type + ", " + std::to_string(NumElems) + ">";
+        log_msg += ". The element at index: " + std::to_string(i) +
+                   ", is not nan, but it should.";
+
+        log::note(log_msg);
+      }
+    }
+    if (are_bitwise_equal(expected, retrieved)) {
+      passed = false;
```

Comment:
```suggestion
    if (!are_bitwise_equal(expected, retrieved)) {
      passed = false;
```
Is it a misprint?

Reply:
Yes, fixed [here](https://github.com/intel/llvm-test-suite/pull/827/commits/7dddeb4523bc5d5bf8bb785d6f17a1c6e2461de4#diff-957362298136f7a2e0ab66c1cff4239caca52ea84dab6e6b2728ccc9411ebe0eR284)